core/runtime/common/runtime_upgrade_tracker_impl.cpp

```diff
@@ -44,11 +60,18 @@ namespace kagome::runtime {
         runtime_upgrade_parents_.emplace_back(block.number - 1,
                                               header.parent_hash);
       }
+      save();
       SL_DEBUG(logger_,
-          "Pick runtime state at block #{} hash {} for the same block",
-          block.number,
-          block.hash.toHex());
-      return header.state_root;
+               "Pick runtime state at block #{} hash {} for the same block",
+               block.number,
+               block.hash.toHex());
+      return block.hash;
+    }
+
+    if (auto it = code_substitutes_.find(block.hash);
```

Comment:
Why is it not enough to just insert then into runtime upgrades during the initialisation?